### PR TITLE
Disable paste for read-only markdown cells & fix replace all for markdown cells

### PR DESCRIPTION
**Pull request by itsmevichu**
Fixes https://github.com/jupyterlab/jupyterlab/issues/11124

And for the issue https://github.com/jupyterlab/jupyterlab/issues/16925, the PR - https://github.com/jupyterlab/jupyterlab/pull/16940 did not address the issue for markdown cells; so, I have addressed those changes too in this PR.

### CODE REVIEW

File: packages/cells/src/searchprovider.ts [hunk 1]
```diff
@@ -429,11 +429,14 @@ class MarkdownCellSearchProvider extends CellSearchProvider {
...
-  async replaceAllMatches(newText: string): Promise<boolean> {
+  async replaceAllMatches(
+    newText: string,
+    options?: IReplaceOptions
+  ): Promise<boolean> {
     if (this.model.getMetadata('editable') === false)
       return Promise.resolve(false);
 
-    const result = await super.replaceAllMatches(newText);
+    const result = await super.replaceAllMatches(newText, options);
```
Comment (krassowski) [inline packages/cells/src/searchprovider.ts:439]: It looks like some changes from PR https://github.com/jupyterlab/jupyterlab/pull/16940 got included in here, let's revert them to have a clean diff :)

File: packages/notebook/test/searchprovider.spec.ts [hunk 2]
```diff
@@ -419,7 +419,7 @@ describe('@jupyterlab/notebook', () => {
       it('should replace all matches using regex', async () => {
         panel.model!.sharedModel.insertCells(0, [
           {
-            cell_type: 'code',
+            cell_type: 'markdown',
```
Comment (krassowski) [inline packages/notebook/test/searchprovider.spec.ts:422]: Same here.

File: packages/cells/src/searchprovider.ts (hunk 1 above)
Comment (krassowski) [reply packages/cells/src/searchprovider.ts:439]: Ah, I see this PR does two things. Make sense.

**Comment by krassowski** [comment]
Hi @itsmevichu I invited you to join the triage team so that you can apply `bug`/`enhancement`/`maintenance` labels on PRs as you see fit :) Welcome!

**Comment by krassowski** [comment]
It looks like `js-cells` test failure is relevant:

```
  ● cells/widget › MarkdownCell › #getEditorOptions() › should normalise line endings on paste

    expect(received).toEqual(expected) // deep equality

    Expected: "
    Test
    String
    .
    "
    Received: ""

      1020 |         widget.editor!.host.querySelector('.cm-content')!.dispatchEvent(event);
      1021 |
    > 1022 |         expect(widget.model.sharedModel.getSource()).toEqual(
           |                                                      ^
      1023 |           '\nTest\nString\n.\n'
      1024 |         );
      1025 |       });

      at Object.<anonymous> (test/widget.spec.ts:1022:54)


Error: expect(received).toEqual(expected) // deep equality

Expected: "
Test
String
.
"
Received: ""
```

**Review by krassowski — APPROVED** [review]
Thank you @itsmevichu!

**Comment by jtpio** [comment]
@meeseeksdev please backport to 4.3.x

**Pull request merged**